buildkite/resource_organization_rule.go

```diff
@@ -0,0 +1,370 @@
+	panic("cannot update an organization rule")
+}
+
+func (or *organizationRuleResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var state organizationRuleResourceModel
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	timeout, diags := or.client.timeouts.Delete(ctx, DefaultTimeout)
+	resp.Diagnostics.Append(diags...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	err := retry.RetryContext(ctx, timeout, func() *retry.RetryError {
+		org, err := or.client.GetOrganizationID()
+		if err == nil {
+			_, err = deleteOrganizationRule(
+				ctx,
+				or.client.genqlient,
+				*org,
+				state.ID.ValueString(),
+			)
+		}
+
+		return retryContextError(err)
+	})
+
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to delete organization rule",
+			fmt.Sprintf("Unable to delete organization rule: %s", err.Error()),
+		)
+		return
+	}
+}
+
+func obtainCreationUUIDs(r *createOrganizationRuleResponse) (*string, *string, error) {
+	var sourceUUID, targetUUID string
+	/*
+		The provider will try and determine the source UUID based on type that is returned in the *createOrganizationRuleResponse
+		It will switch based on the SourceType returned in the response and extract the UUID of the respective source based on this.
+		Otherwise, it will create and throw an error stating that it cannot obtain the source type from the returned API response.
+	*/
+	switch r.RuleCreate.Rule.SourceType {
+	case "PIPELINE":
+		if ruleCreateSourcePipeline, ok := r.RuleCreate.Rule.Source.(*OrganizationRuleFieldsSourcePipeline); ok {
+			sourceUUID = ruleCreateSourcePipeline.Uuid
+		} else {
+			return nil, nil, errors.New("Error obtaining source type upon creating the organization rule.")
+		}
+	}
+
+	/*
+		 	Now, like above - the provider will try and determine the target UUID based on the *createOrganizationRuleResponse. It will
+			switch based on the TargetType returned in the response and extract the UUID of the respective source based on this.
+			Otherwise, it will create and throw an error stating that it cannot obtain the source type from the returned API response.
+	*/
+	switch r.RuleCreate.Rule.TargetType {
+	case "PIPELINE":
+		if ruleCreateTargetPipeline, ok := r.RuleCreate.Rule.Target.(*OrganizationRuleFieldsTargetPipeline); ok {
+			targetUUID = ruleCreateTargetPipeline.Uuid
+		} else {
+			return nil, nil, errors.New("Error obtaining target type upon creating the organization rule.")
+		}
+	}
+
+	return &sourceUUID, &targetUUID, nil
+}
+
+func obtainReadUUIDs(nr getNodeNodeRule) (string, string) {
+	var sourceUUID, targetUUID string
+
+	switch nr.SourceType {
+	case "PIPELINE":
+		sourceUUID = nr.Source.(*OrganizationRuleFieldsSourcePipeline).Uuid
+	}
+
+	switch nr.TargetType {
+	case "PIPELINE":
+		targetUUID = nr.Target.(*OrganizationRuleFieldsTargetPipeline).Uuid
+	}
+
+	return sourceUUID, targetUUID
```

Comment:
This should be fine as it will essentially return `""`

Reply:
👍👍